cmd/root.go

```diff
@@ -68,11 +78,84 @@ func init() {
 		"logging level; one of [trace, debug, info, warning, error, fatal]")
 }
 
-func sudoCheck(_ *cobra.Command, _ []string) error {
-	id := os.Geteuid()
-	if id != 0 {
-		return errors.New("containerlab requires sudo privileges to run")
+func checkAndGetRootPrivs(_ *cobra.Command, _ []string) error {
+	_, euid, suid := unix.Getresuid()
+	if euid != 0 && suid != 0 {
+		return fmt.Errorf("this containerlab command requires root privileges or root via SUID to run, effective UID: %v SUID: %v", euid, suid)
+	}
+
```

Comment:
@vista- in https://github.com/srl-labs/containerlab/pull/2420 a light refactoring has been done to the version command which is now in its own package under `cmd`

Also, the sudoCheck moved to the `cmd/common` package. If you rebase this PR on the `main` and align with these changes it would be fantastic

Reply:
Rebased PR on the refactor